packages/@o3r/rules-engine/src/engine/operator/operator.helpers.ts

```diff
@@ -7,16 +8,28 @@ import {DateInput, Operator, SupportedSimpleTypes} from './operator.interface';
  * @param rhs Right hand side
  * @param operator Operator to compare values
  */
-export function executeOperator<L = unknown, R = unknown>(lhs: L, rhs: R, operator: Operator<L, R>) {
+export function executeOperator<L = unknown, R = unknown>(lhs: L, rhs: R, operator: Operator<L, R>, operatorFacts?: Record<string, Facts | undefined>) {
   const validLhs = (!operator.validateLhs || operator.validateLhs(lhs));
   const validRhs = (!operator.validateRhs || operator.validateRhs(rhs));
+  let operatorFactValues: Record<string, Facts> | undefined;
+  if (operatorFacts && operator.dependencies) {
+    operatorFactValues = operator.dependencies.reduce((acc, dep) => {
+      if (operatorFacts[dep]) {
```

Comment:
this is exactly the purpose of the `validateOperatorFactValues` on the next line

Reply:
as discussed, this should be the job of the operatorExecutor, as the list of dependencies has been already provided by the operator.
The operators can check the value emitted by the fact but it does not need to check it existence itself

Reply:
We cannot throw in the `else` because in `operatorFacts` we have all the input facts. The goal here is to execute `operator.evaluator` only if we have valid values for `lhs` `rhs`  and `operatorFactValues` (based on the operator's dependencies)